api/v1alpha1/dnsrecord_types.go

```diff
@@ -33,10 +33,19 @@ const HttpsProtocol HealthProtocol = "HTTPS"
 // By default this health check will be applied to each unique DNS A Record for
 // the listeners assigned to the target gateway
 type HealthCheckSpec struct {
-	Endpoint         string          `json:"endpoint,omitempty"`
-	Port             *int            `json:"port,omitempty"`
-	Protocol         *HealthProtocol `json:"protocol,omitempty"`
-	FailureThreshold *int            `json:"failureThreshold,omitempty"`
+	// Should start with "?" or "|", contain only allowed for URLs characters and end with alphanumeric char or "/"
+	// +kubebuilder:validation:Pattern=`^(?:\?|\/)[\w\-.~:\/?#\[\]@!$&'()*+,;=]+(?:[a-zA-Z0-9]|\/){1}$`
+	Endpoint string `json:"endpoint,omitempty"`
+
+	// Only following ports allowed: 80, 443, 1024-49151
+	// +kubebuilder:validation:XValidation:rule="self in [80, 443] || (self >= 1024 && self <= 49151)",message="Allow ports 80, 443, 1024-49151 are allowed"
+	Port *int `json:"port,omitempty"`
+
+	// +kubebuilder:validation:XValidation:rule="self in ['HTTP','HTTPS']",message="Only HTTP or HTTPS protocols are allowed"
```

Comment:
is this case sensitive? Ideally it wouldn't be.

Reply:
Right now if we will use lower case protocols the dns-operator will break. Well, not break but will try to create healtcheck with `nil` protocol. I can change that to allow for `http` and `https` ports. but that would mean changing the logic a bit :thinking: